src/spatio_temporal_voxel_layer.cpp

```diff
@@ -513,7 +513,7 @@ void SpatioTemporalVoxelLayer::activate(void)
 /*****************************************************************************/
 {
   // subscribe and place info in buffers from sensor sources
-  RCLCPP_INFO(node_->get_logger(), "%s was activated.", getName().c_str());
+  RCLCPP_INFO((node_.lock())->get_logger(), "%s was activated.", getName().c_str());
```

Comment:
in `onInitialize` you should store the logger object while the shared pointer is locked as a class member and change all these log statements to using that logger object